proposal.md

```diff
@@ -1,20 +1,58 @@
+I would like to create a fault tolerant web based multiplayer game.  The 'business logic' will be
+taken care of by the elixir backend and the frontend will handle user input/rendering.  The game will render
+onto a canvas element.  User -> Server communication will be done using web sockets (seems neccesary as the game will be real time).
+
+The basic structure of the game is a 2d shooter in the general "Style" of <a href="agar.io">agar.io</a> (read: simple).
+The primary goal of this project is to get down the backend server structure, client communication, and effective state management.
+
+I think the potential scope of this is quite large so while I plan on completing the entire game (including the web app),
+I will work in an order that lends itself to having something to showcase at the end of the project regardless of how far I get.  
+
+To cover the case that getting a functional backend is a larger time commitment than expected, I will implement the
+'business logic' first.  If it seems realistic to also get a web client done and all communication I will proceed in that direction - otherwise I will work on an elixir simulator to test the game engine.  The entire client api will be exposed via a GenServer api so I will be able to write a fully functional demo of the engine and render the results to a terminal or some sort of alternative display.
+
+That being said the whole reason I picked this project over a data visualization is I thought it would be fun to make a highly fault tolerant web game - so I am really leaning that direction even if it takes longer than rendering a simulation locally to a terminal.
 
 ## Outline Structure
 
-> Describe how you'll organize your code. What is the process and
-> supervision structure? If it uses a framework, how does it fit in. I
-> just need to understand the direction you plan to take. Again, three
-> or four sentences are probably enough.
+I've begun to work a bit on the project and I currently have this module structure:
+
+2DShooterPhoenixServer
+-> this will have one or multiple GameServers running on it (last priority is implementing support for concurrent games.)
+
+GameServer
+-> has/supervises multiple child processes
+-> I believe the game server will initially be :permanent in the case of a single game instance, :temporal if I add support for concurrent games.
+-> handles user input and delegates that input to respective servers
+-> handles system time changes and passes that information down to children to maintain synchronization across children
+-> uses Task.async to simultaneously update it's children on each call to "tick"
+-> BulletServer
+-> PlayerServer
+-> still deciding where collision detection should take place.
+
+and the other servers:
+
+BulletServer
+-> maintains the status of all currently existing bullets
+-> cleans bullets out after their "expiration" time.
+-> updates bullets appropriately based on the time passed from the parent
```

Comment:
If each bullet is a separate process, it can just kill itself. If can also do its own position update etc.